### ISSUE TITLE
Flutter material icons not showing in deploying as release version

### ISSUE THREAD

**Issue by AbdelahmanZimozi** [flutter/flutter#143508]
### Steps to reproduce

<img width="785" alt="Screenshot 2024-02-15 at 5 54 30 AM" src="https://github.com/flutter/flutter/assets/130447366/eeabee38-7062-4cfa-b8c2-019471fa3373">
        - flutter clean
        - flutter pub get
        - flutter build web --web-renderer canvaskit --release --dart-define=FLUTTER_WEB_USE_SKIA=true --dart-define=FLUTTER_WEB_USE_EXPERIMENTAL_CANVAS_TEXT=true --no-tree-shake-icons

### Expected results

see flutter icons like local host 

### Actual results

I don't see any flutter  icons see boxes 

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>
<img width="785" alt="Screenshot 2024-02-15 at 5 54 30 AM" src="https://github.com/flutter/flutter/assets/130447366/6f793816-98dd-4739-8f85-9a396fc99b5a">

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @AbdelahmanZimozi, please provide the output of `flutter doctor -v`. 

Where is your website being hosted? Do you see any errors in your web browser console?

**Comment by AbdelahmanZimozi** [comment]
<img width="1420" alt="Screenshot 2024-02-15 at 12 28 06 PM" src="https://github.com/flutter/flutter/assets/130447366/db9267e8-3e6c-4c65-a747-1e222f38a36e">
I am getting Failed to load font MaterialIcons at assets/fonts/MaterialIcons-Regular.otf

**Comment by AbdelahmanZimozi** [comment]
flutter doctor -v                                                 
[!] Flutter (Channel stable, 3.16.9, on macOS 14.0 23A344 darwin-arm64, locale en-EG)
    • Flutter version 3.16.9 on channel stable at /Users/abdelrahmanahmed/flutter
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/3.2.2/libexec/bin/dart, which is not inside your current Flutter SDK checkout at
      /Users/abdelrahmanahmed/flutter. Consider adding /Users/abdelrahmanahmed/flutter/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 41456452f2 (3 weeks ago), 2024-01-25 10:06:23 -0800
    • Engine revision f40e976bed
    • Dart version 3.2.6
    • DevTools version 2.28.5
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] IntelliJ IDEA Community Edition (version 2023.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 75.1.4
    • Dart plugin version 232.9559.10

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 14.0 23A344 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 121.0.6167.160

[✓] Network resources
    • All expected network resources are available.

**Comment by AbdelahmanZimozi** [comment]
Where is your website being hosted? AWS amplify to hosting

**Comment by AbdelahmanZimozi** [comment]
![image](https://github.com/flutter/flutter/assets/130447366/6b826c0b-abb8-452e-b7d9-89bdf2643d1e)

**Comment by AbdelahmanZimozi** [comment]
@danagbemava-nc  Do you know what's the problem ? , how can solve it ?

**Comment by danagbemava-nc** [comment]
Hi @AbdelahmanZimozi, this has come up several times before but we haven't been able to fix it because we haven't been able to reproduce the issue ourselves. Can you try a different hosting service, like firebase hosting or netlify to see if this still reproduces for you? 


Can you also try using `flutter build web --release` instead of `flutter build web --web-renderer canvaskit --release --dart-define=FLUTTER_WEB_USE_SKIA=true --dart-define=FLUTTER_WEB_USE_EXPERIMENTAL_CANVAS_TEXT=true --no-tree-shake-icons` to build the app to see if you still experience this issue? 

Lastly, a new stable version was released yesterday (3.19.0), can you try upgrading to that version and trying again? 


Similar issues: 
https://github.com/flutter/flutter/issues/68704
https://github.com/flutter/flutter/issues/32540
https://github.com/flutter/flutter/issues/68704
https://github.com/flutter/flutter/issues/130000

**Comment by inzywong** [comment]
I have this issue to, commenting to be notified if solution is found :)
